src/algoliasearch.helper.js

```diff
@@ -431,10 +439,184 @@ AlgoliaSearchHelper.prototype.setState = function(newState) {
+ *  - filters : possible values are all the keys of the {SearchParameters}, 'index' for the index,
+ *    all the refinements with 'attribute:*' or for some specific attributes with 'attribute:theAttribute'
+ *  - prefix : prefix in front of the keys
+ *  - moreAttributes : more values to be added in the query string. Those values
+ *    won't be prefixed.
+ * @return {string} the query string
+ */
+AlgoliaSearchHelper.prototype.getStateAsQueryString = function getStateAsQueryString(options) {
+  var filters = options && options.filters || ['query', 'attribute:*'];
+  var moreAttributes = options && options.moreAttributes;
+  var prefixForParameters = options && options.prefix || '';
+
+  var partialState = this.getState(filters);
+
+  var encodedState = mapKeys(
+    partialState,
+    function(v, k) {
+      var shortK = shortener.encode(k);
+      return prefixForParameters + shortK;
+    }
+  );
+
+  if (moreAttributes) merge(encodedState, moreAttributes);
+
+  return qs.stringify(encodedState, {encode: false});
+};
+
+/**
+ * Overrides part of the state with the properties stored in the provided query
+ * string.
+ * @param {string} queryString the query string containing the informations to url the state
+ * @param {object} options optionnal parameters :
+ *  - prefix : prefix used for the algolia parameters
+ *  - triggerChange : if set to true the state update will trigger a change event
  */
-AlgoliaSearchHelper.prototype.getState = function() {
-  return this.state;
+AlgoliaSearchHelper.prototype.setStateFromQueryString = function setStateFromQueryString(queryString, options) {
+  var triggerChange = options && options.triggerChange || false;
+
+  var configuration = AlgoliaSearchHelper.getConfigurationFromQueryString(queryString, options);
+
+  var updatedState = this.state.setQueryParameters(configuration);
+
+  if (triggerChange) this.setState(updatedState);
+  else this.overrideStateWithoutTriggeringChangeEvent(updatedState);
+};
+
+/**
+ * Read a query string and return an object containing the state
+ * @static
+ * @param {string} queryString the query string that will be decoded
+ * @param {object} options accepted options :
+ *   - prefix : the prefix used for the saved attributes, you have to provide the
+ *     same that was used for serialization
+ * @return {object} partial search parameters object (same properties than in the
+ * SearchParameters but not exhaustive
+ */
+AlgoliaSearchHelper.getConfigurationFromQueryString = function(queryString, options) {
+  var prefixForParameters = options && options.prefix || '';
+
+  var partialStateWithPrefix = qs.parse(queryString);
+  var prefixRegexp = new RegExp('^' + prefixForParameters);
+  var partialState = mapKeys(
+    partialStateWithPrefix,
+    function(v, k) {
+      if (prefixForParameters && prefixRegexp.test(k)) {
+        var encodedKey = k.replace(prefixRegexp, '');
+        return shortener.decode(encodedKey);
```

Comment:
If the key match the prefix (which is supposed to be a namespace for our keys) and is not decodable, it means that the prefix was not good. What about we throw an error?